docs/utilities/idempotency.md

```diff
@@ -97,66 +124,80 @@ If you're not [changing the default configuration for the DynamoDB persistence l
+
 ### Idempotent decorator
 
-You can quickly start by initializing the `DynamoDBPersistenceLayer` class and using it with the `idempotent` decorator on your lambda handler.
+For simple use cases, you can use the `idempotent` decorator on your Lambda handler function.
 
-???+ note
-    In this example, the entire Lambda handler is treated as a single idempotent operation. If your Lambda handler can cause multiple side effects, or you're only interested in making a specific logic idempotent, use [`idempotent_function`](#idempotent_function-decorator) instead.
-
-!!! tip "See [Choosing a payload subset for idempotency](#choosing-a-payload-subset-for-idempotency) for more elaborate use cases."
+It will treat the entire event as an idempotency key. That is, the same event will return the previously stored result within a [configurable time window](#adjusting-expiration-window) _(1 hour, by default)_.
 
 === "Idempotent decorator"
 
-    ```python hl_lines="4-7 10 24"
+    !!! tip "You can also choose [one or more fields](#choosing-a-payload-subset) as an idempotency key."
+
+    ```python title="getting_started_with_idempotency.py" hl_lines="5-8 12 25"
     --8<-- "examples/idempotency/src/getting_started_with_idempotency.py"
     ```
 
 === "Sample event"
 
-    ```json
+    ```json title="getting_started_with_idempotency_payload.json"
     --8<-- "examples/idempotency/src/getting_started_with_idempotency_payload.json"
     ```
 
-After processing this request successfully, a second request containing the exact same payload above will now return the same response, ensuring our customer isn't charged twice.
-
-!!! question "New to idempotency concept? Please review our [Terminology](#terminology) section if you haven't yet."
-
 ### Idempotent_function decorator
 
-Similar to [idempotent decorator](#idempotent-decorator), you can use `idempotent_function` decorator for any synchronous Python function.
+For full flexibility, you can use the `idempotent_function` decorator for any synchronous Python function.
 
-When using `idempotent_function`, you must tell us which keyword parameter in your function signature has the data we should use via **`data_keyword_argument`**.
+When using this decorator, you **must** call your decorated function using keyword arguments.
 
-!!! tip "We support JSON serializable data, [Python Dataclasses](https://docs.python.org/3.12/library/dataclasses.html){target="_blank" rel="nofollow"}, [Parser/Pydantic Models](parser.md){target="_blank"}, and our [Event Source Data Classes](./data_classes.md){target="_blank"}."
-
-???+ warning "Limitation"
-    Make sure to call your decorated function using keyword arguments.
+You can use `data_keyword_argument` to tell us the argument to extract an idempotency key.  We support JSON serializable data, [Dataclasses](https://docs.python.org/3.12/library/dataclasses.html){target="_blank" rel="nofollow"}, Pydantic Models, and [Event Source Data Classes](./data_classes.md){target="_blank"}
```

Comment:
```suggestion
You can use `data_keyword_argument` to tell us the argument to extract an idempotency key.  We support JSON serializable data, [Dataclasses](https://docs.python.org/3.12/library/dataclasses.html){target="_blank" rel="nofollow"}, Pydantic Models, and [Event Source Data Classes](./data_classes.md){target="_blank"}.
```

Reply:
We support to serialize other type of data using CustomDictSerializer, no?

Reply:
not here, because it's an input (`fn(order: Order)`). Which is a **GREAT** catch because we're missing that functionality today -- similar to Event Handler we should allow customers to bring a custom serializer for input in case they want to bring in a binary of sorts and convert it to JSON before storing in the persistent storage.

`CustomDictSerializer` is for the **output**, which we document under `Any type`

<img width="1160" alt="image" src="https://github.com/user-attachments/assets/860abec9-1365-4274-bf7f-eaa6203c43ae">